### DISCUSSION TITLE
Async code in ModbusBaseSlaveContext

### DISCUSSION THREAD

**Discussion by ilkka-ollakka** [pymodbus-dev/pymodbus#2104] [Q&A] [answered]
Is there some example or API that shows how to implement async ModbusBaseSlaveContext?

I have looked the current RemoteSlaveContext and contribs, but all seem to use sync code. So is there some flow that I have missed to use async code in there, or is there some implementation that I should do myself somewhere else in the modbus-server flow to be able to use async code?

**Comment by janiversen** [comment, marked as answer]
The current context is sync, we are slowly working towards making it async.
